aci/resource_aci_fvnsmcastaddrinstp.go

```diff
@@ -0,0 +1,307 @@
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	fvnsMcastAddrInstP, err := getRemoteMulticastAddressPool(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	_, err = setMulticastAddressPoolAttributes(fvnsMcastAddrInstP, d)
+	if err != nil {
+		return nil, err
+	}
+
+	multicastAddressBlocks := getMulticastAddressBlocks("Import", fvnsMcastAddrInstP.Name, aciClient, d)
+	if multicastAddressBlocks != nil {
+		setMulticastAddressBlocks("Import", fvnsMcastAddrInstP.Name, multicastAddressBlocks, aciClient, d)
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceAciMulticastAddressPoolCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] MulticastAddressPool: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	name := d.Get("name").(string)
+	fvnsMcastAddrInstPAttr := models.MulticastAddressPoolAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		fvnsMcastAddrInstPAttr.Annotation = Annotation.(string)
+	} else {
+		fvnsMcastAddrInstPAttr.Annotation = "{}"
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		fvnsMcastAddrInstPAttr.Name = Name.(string)
+	}
+
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		fvnsMcastAddrInstPAttr.NameAlias = NameAlias.(string)
+	}
+	fvnsMcastAddrInstP := models.NewMulticastAddressPool(fmt.Sprintf(models.RnfvnsMcastAddrInstP, name), models.ParentDnfvnsMcastAddrInstP, desc, fvnsMcastAddrInstPAttr)
+	err := aciClient.Save(fvnsMcastAddrInstP)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if multicastAddressBlock, ok := d.GetOk("multicast_address_block"); ok {
+		multicastAddressBlockList := multicastAddressBlock.(*schema.Set).List()
+		multicastAddressBlocks := make([]*models.MulticastAddressBlock, len(multicastAddressBlockList))
+		for index, block := range multicastAddressBlockList {
+			blockMap := block.(map[string]interface{})
+			fvnsMcastAddrBlkAttr := models.MulticastAddressBlockAttributes{}
+			fvnsMcastAddrBlkAttr.Annotation = blockMap["annotation"].(string)
+			fvnsMcastAddrBlkAttr.From = blockMap["from"].(string)
+			fvnsMcastAddrBlkAttr.Name = blockMap["name"].(string)
+			fvnsMcastAddrBlkAttr.To = blockMap["to"].(string)
+			if blockMap["nameAlias"] != nil {
+				fvnsMcastAddrBlkAttr.NameAlias = blockMap["nameAlias"].(string)
+			}
+			desc := ""
+			if blockMap["description"] != nil {
+				desc = blockMap["description"].(string)
+			}
+			fvnsMcastAddrBlk := models.NewMulticastAddressBlock(fmt.Sprintf(models.RnfvnsMcastAddrBlk, fvnsMcastAddrBlkAttr.From, fvnsMcastAddrBlkAttr.To), fvnsMcastAddrInstP.DistinguishedName, desc, fvnsMcastAddrBlkAttr)
+			err := aciClient.Save(fvnsMcastAddrBlk)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			multicastAddressBlocks[index] = fvnsMcastAddrBlk
```

Comment:
Any reason to use a fixed length list and not use something like append?

Reply:
done also for update